bugwarrior/services/logseq.py

```diff
@@ -0,0 +1,301 @@
+                f"http://{self.host}:{self.port}/api",
+                headers=self.headers,
+                json={"method": "logseq.DB.datascriptQuery", "args": [query]},
+            )
+            return self.json_response(response)
+        except requests.exceptions.ConnectionError as ce:
+            log.fatal("Unable to connect to Logseq HTTP APIs server. %s", ce)
+            exit()
+
+    def _get_current_graph(self):
+        try:
+            response = requests.post(
+                f"http://{self.host}:{self.port}/api",
+                headers=self.headers,
+                json={"method": "logseq.getCurrentGraph", "args": []},
+            )
+            return self.json_response(response)
+        except requests.exceptions.ConnectionError as ce:
+            log.fatal("Unable to connect to Logseq HTTP APIs server. %s", ce)
+            exit()
+
+    def get_graph_name(self):
+        graph = self._get_current_graph()
+        return graph["name"] if graph else None
+
+    def get_issues(self):
+        task_filter = self.filter.replace(" ", "").replace(",", '" "')
+        return self._datascript_query(
+            f"""
+            [:find (pull ?b [*])
+                :where [?b :block/marker ?marker]
+                [(contains? #{{\"{task_filter}\"}} ?marker)]
+            ]
+        """
+        )
+
+
+class LogseqIssue(Issue):
+    ID = "logseqid"
+    UUID = "logsequuid"
+    STATE = "logseqstate"
+    TITLE = "logseqtitle"
+    SCHEDULED = "logseqscheduled"
+    DEADLINE = "logseqdeadline"
+    DONE = "logseqdone"
+    URI = "logsequri"
+
+    # Local 2038-01-18, with time 00:00:00.
+    # A date far away, with semantically meaningful to GTD users.
+    # see https://taskwarrior.org/docs/dates/
+    SOMEDAY = datetime(2038, 1, 18)
+
+    UDAS = {
+        ID: {
+            "type": "string",
+            "label": "Logseq ID",
+        },
+        UUID: {
+            "type": "string",
+            "label": "Logseq UUID",
+        },
+        STATE: {
+            "type": "string",
+            "label": "Logseq State",
+        },
+        TITLE: {
+            "type": "string",
+            "label": "Logseq Title",
+        },
+        SCHEDULED: {
+            "type": "date",
+            "label": "Logseq Scheduled",
+        },
+        DEADLINE: {
+            "type": "date",
+            "label": "Logseq Deadline",
+        },
+        DONE: {
+            "type": "date",
+            "label": "Logseq Done",
+        },
+        URI: {
+            "type": "string",
+            "label": "Logseq URI",
+        },
+    }
+
+    UNIQUE_KEY = (ID, UUID)
+
+    # map A B C priority to H M L
+    PRIORITY_MAP = {
+        "A": "H",
+        "B": "M",
+        "C": "L",
+    }
+
+    STATE_MAP = {
+        "IN-PROGRESS": "pending",
+        "DOING": "pending",
+        "TODO": "pending",
+        "NOW": "pending",
+        "LATER": "waiting",
+        "WAIT": "waiting",
+        "WAITING": "waiting",
```

Comment:
Per https://github.com/GothenburgBitFactory/taskwarrior/pull/2563, I'm now under the impression that the status should never be manually set to `waiting`. Given that, I'm surprised it worked. @djmitche Can you confirm that I'm understanding this correctly?

Reply:
Updated the logic to use wait date rather that waiting status.